api/test/integration/indicator-record/indicator-records.service.spec.ts

```diff
@@ -0,0 +1,364 @@
+      jest.spyOn(materialsToH3sService, 'findOne').mockResolvedValue({
+        materialId: UUIDv4(),
+        id: UUIDv4(),
+        h3DataId: UUIDv4(),
+        ...({} as any),
+      });
+      jest
+        .spyOn(
+          indicatorRecordRepository,
+          'getIndicatorRawDataByGeoRegionAndMaterial',
+        )
+        .mockResolvedValue({
+          production: 10,
+          harvestedArea: 100,
+          rawBiodiversity: 10,
+          rawCarbon: 5,
+          rawDeforestation: 20,
+          rawWater: 25,
+        });
+
+      //ACT
+      await indicatorRecordService.createIndicatorRecordsBySourcingRecords(
+        sourcingData,
+      );
+
+      //ASSERT
+      const allIndicators = await indicatorRecordRepository.find();
+      expect(allIndicators.length).toEqual(4);
+
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.DEFORESTATION,
+        indicatorPreconditions.deforestation,
+        sourcingData,
+        1000,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.BIODIVERSITY_LOSS,
+        indicatorPreconditions.biodiversityLoss,
+        sourcingData,
+        500,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.CARBON_EMISSIONS,
+        indicatorPreconditions.carbonEmissions,
+        sourcingData,
+        250,
+        10,
+      );
+      await checkCreatedIndicatorRecord(
+        INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+        indicatorPreconditions.waterRisk,
+        sourcingData,
+        12500,
+        10,
+      );
+    });
+  });
+
+  async function checkCreatedIndicatorRecord(
+    indicatorType: INDICATOR_TYPES,
+    h3Data: H3Data,
+    sourcingData: any,
+    recordValue: number,
+    scalerValue: number | null,
+  ): Promise<void> {
+    const indicatorRecords = await indicatorRecordRepository.find({
+      where: { indicatorId: h3Data.indicatorId },
+    });
+    expect(indicatorRecords.length).toEqual(1);
+    expect(indicatorRecords[0].sourcingRecordId).toEqual(
+      sourcingData.sourcingRecordId,
+    );
+    expect(indicatorRecords[0].status).toEqual(INDICATOR_RECORD_STATUS.SUCCESS);
+    expect(indicatorRecords[0].value).toEqual(recordValue);
+    expect(indicatorRecords[0].scaler).toEqual(scalerValue);
+    expect(indicatorRecords[0].h3DataId).toEqual(h3Data.id);
+    //Inidicator Coefficientes are not checked because it's not used
+  }
+
+  async function deleteH3GridTables(): Promise<void> {
+    const tableList: any[] = await getManager().query(`select table_name
+                                                       from information_schema.tables
+                                                       where table_name like '%h3_grid_%'`);
+    if (tableList.length) {
+      const tableNameList = tableList.map((table: any) => table.table_name);
+      await getManager().query(`DROP TABLE ${tableNameList.join(', ')}`);
+    }
+  }
+
+  async function createPreconditions(): Promise<any> {
```

Comment:
Same for this, do you think it would be useful to have this available in a broader context, or do you think it's tied to this suite?

If it's tied to this context, maybe it will be a good idea to have it in a external file, what do you think?